ms/MeasurementSets/MSIter.cc

```diff
@@ -361,6 +361,7 @@ void MSIter::origin()
 {
   curMS_p=0;
   checkFeed_p=True;
+  if (timeComp_p) timeComp_p->setOffset(0.0);

```

Comment:
Sure, but the situation is far more subtle:  The setState() call in origin() comes _after_ a call to TableIterator::reset, which depends upon the internal state of timeComp_p.   If a 2nd pass through the iterator is done---i.e., a 2nd call to MSIter::origin()---the internal state of timeComp_p (seen by TableIterator::reset) is unpredictable, as it depends on the value of the offset_p at the end of the previous pass.  Note that setState() is used both in MSIter::origin() and in MSIter::advance(), mainly to set some meta-info about the current iteration.  IMO, it was quite sloppy to hide the timeComp_p reset in setState(), as timeComp_p is part of managing the behavior of the underlying iterator, not current iteration meta-info.
HOWEVER, this change was a legacy of testing variation in the MSIter specialization I am forming in CASA, and since I do not want to change observable behavior in MSIter with this revision, I should not have included it here.  I will remove this change and re-implement it in my specialization.  This will require MSIter::origin() to be virtual....